### DISCUSSION TITLE
How does one find the size of a deku binary representation (for fixed size)

### DISCUSSION THREAD

**Discussion by ghost** [sharksforarms/deku#511] [Q&A]
```rust
#[derive(DekuRead, DekuWrite, Debug, PartialEq)]
#[deku(endian = "big")]
struct AifsLayer1Header {
    magic: u32,
    size: u64,
    level: u8,
    reserved: [u8; 1000],
}
```

suppose this is the struct

how do i get the size of the struct's binary representation in number of bytes.

**Comment by wcampbell0x2a** [comment]
For that struct, since you don't have deku attributes on the fields and everything's standard, you can use [mem::size_of](https://doc.rust-lang.org/std/mem/fn.size_of.html).

If you start using deku attributes on the fields, there is an open issue for that feature: https://github.com/sharksforarms/deku/issues/307

**Reply by ghost** [reply]
ty!! but won't the alignment of struct cause issues?
